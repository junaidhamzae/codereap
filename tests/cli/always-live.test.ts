import path from 'node:path';
import fs from 'node:fs';
import os from 'node:os';
import { runCli } from '../helpers/cli';

describe('CLI --alwaysLive globs', () => {
  it('marks files live without edges', () => {
    const fixture = path.resolve(__dirname, '../fixtures/always-live');
    const tmp = fs.mkdtempSync(path.join(os.tmpdir(), 'cli-'));
    const outBase = path.join(tmp, 'report');
    const res = runCli(['--root', fixture, '--out', outBase, '--alwaysLive', '**/E.*']);
    expect(res.status).toBe(0);
    const rows = res.json as any[];
    const e = rows.find(r => r.node.endsWith('E.ts'))!;
    expect(e.orphan).toBe(false);
    expect(rows).toMatchSnapshot();
  });
});


