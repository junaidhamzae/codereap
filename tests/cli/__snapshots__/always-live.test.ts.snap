// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CLI --alwaysLive globs marks files live without edges 1`] = `
[
  {
    "exists": true,
    "in-degree": 0,
    "node": "A.ts",
    "orphan": true,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [],
        "reExports": [],
      },
      "imports": [],
    },
  },
  {
    "exists": true,
    "in-degree": 0,
    "node": "E.ts",
    "orphan": false,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [
          {
            "name": "E",
            "orphan": true,
            "reexport": false,
            "referencedInFile": false,
          },
        ],
        "reExports": [],
      },
    },
  },
]
`;
